MPM/mpm.php

```diff
@@ -96,8 +96,8 @@ function mpm_webhook()
 		$order_id	= $payment->metadata->order_id;
 		$order		= new WC_Order( $order_id );
 
-		// Don't update an already paid order
-		if (!$order->needs_payment())
+		// Don't update an already paid order, needs_payment() searches by valid_statusses, but does not include on-hold status, so we add it here.
+		if (!$order->needs_payment() && $order->post->post_status != 'wc-on-hold')
```

Comment:
`$order->post` bestaat bij mij niet in WooCommerce 2.1.10. Wel heb ik `$order->status = 'on-hold'`. Laatste versie lijkt dit inderdaad via `$order->post` te doen. Je kunt dan ook gebruik maken van `$order->get_status()`. https://github.com/woothemes/woocommerce/blob/master/includes%2Fabstracts%2Fabstract-wc-order.php#L880
Deze methode is helaas niet in mijn versie beschikbaar.